extensions/sql-database-projects/src/tools/newProjectTool.ts

```diff
@@ -0,0 +1,89 @@
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as fs from 'fs';
+import * as os from 'os';
+import * as nls from 'vscode-nls';
+import * as path from 'path';
+import * as constants from '../common/constants';
+
+const localize = nls.loadMessageBundle();
+const MaxDefaultProjectNameCounter: number = 99;
+
+export const DBProjectConfigurationKey: string = 'sqlDatabaseProjects';
+export const ProjectSaveLocationKey: string = 'defaultProjectSaveLocation';
+export const NewDefaultProjectSaveLocation: string = localize('sqlDatabaseProjects.newDefaultProjectSaveLocation', "Would you like to set the default location to save new Database Projects?");
+export const InvalidDefaultProjectSaveLocation: string = localize('sqlDatabaseProjects.invalidDefaultProjectSaveLocation', "Default location to save new Database Projects is invalid. Would you like to update it?");
+export const OpenWorkspaceSettings: string = localize('sqlDatabaseProjects.openWorkspaceSettings', "Yes, open Settings");
+
+export class NewProjectTool {
+	/* Returns the default location to save a new database project*/
+	public get defaultProjectSaveLocation(): vscode.Uri {
+		return this.projectSaveLocationSettingIsValid ? vscode.Uri.file(this.projectSaveLocationSetting) : vscode.Uri.file(os.homedir());
+	}
+
+	/* Returns the workspace setting on the default location to save new database projects*/
+	private get projectSaveLocationSetting(): string {
+		return vscode.workspace.getConfiguration(DBProjectConfigurationKey)[ProjectSaveLocationKey];
+	}
+
+	/* Returns if the default save location for new database projects workspace setting exists and is
+		a valid path*/
+	private get projectSaveLocationSettingIsValid(): boolean {
+		return this.projectSaveLocationSettingExists && fs.existsSync(this.projectSaveLocationSetting);
+	}
+
+	/* Returns if a value for the default save location for new database projects exists*/
+	private get projectSaveLocationSettingExists(): boolean {
+		return this.projectSaveLocationSetting !== null && this.projectSaveLocationSetting !== undefined
+			&& this.projectSaveLocationSetting.trim() !== '';
+	}
+
+	/* Returns default project name for a fresh new project, such as 'DatabaseProject1'. Auto-increments
+		the suggestion if a project of that name already exists in the default save location */
+	public defaultProjectNameNewProj(): string {
+		return this.defaultProjectName(constants.defaultProjectNameStarter, 1);
+	}
+
+	/* Returns default project name for a new project based on given dbName. Auto-increments
+		the suggestion if a project of that name already exists in the default save location */
+	public defaultProjectNameFromDb(dbName: string): string {
+		let projectNameStarter = constants.defaultProjectNameStarter + dbName;
```

Comment:
it's a good habit to use const instead of let if the variable isn't going to be changed